src/modules/server-side/getProfile.ts

```diff
@@ -88,3 +94,64 @@ export const getPropsForWalletOrUsername: GetServerSideProps<WalletDependantPage
     },
   };
 };
+
+export const getProfileServerSideProps: GetServerSideProps<WalletDependantPageProps> = async (
+  ctx
+) => {
+  // Keeping this out of apollo since this isn't modified in-app.
+  const { getProfileInfoFromPubKey, getProfileInfoFromTwitterHandle } =
+    getSdk(graphqlRequestClient);
+
+  const input = ctx.params?.publicKey;
+
+  if (typeof input !== 'string') {
+    return { notFound: true };
+  }
+
+  let publicKey = getPublicKey(input);
+
+  let profileInfo: {
+    walletAddress?: string | null;
+    handle: string;
+    profileImageUrl: string;
+    bannerImageUrl: string;
+  } | null = null;
+  if (publicKey) {
+    const result = await getProfileInfoFromPubKey({ pubKey: publicKey.toBase58() });
+    profileInfo = result?.wallet?.profile ?? null;
+  } else if (isTwitterUsername(input)) {
+    const result = await getProfileInfoFromTwitterHandle({ handle: input });
+    profileInfo = result?.profile ?? null;
+    if (profileInfo?.walletAddress) {
+      publicKey = getPublicKey(profileInfo.walletAddress);
```

Comment:
We chatted at standup and think you are adjusting but I'm worried about page load performance with querying so much on the server. Server side queries should be reserved for data we need to decide on routing choices, denials, and meta head tag info.